### PR TITLE
enable versioning for docs site

### PR DESCRIPTION
**Pull request by geoffcline**
**1. Issue, if available:**
#800 

**2. Description of changes:**
edit: 

- add new dropdown menu for versions
- add new directory for pre-release docs
- add listing of versions to conf.toml
- add script to insert warning in old/prerelease versions

**3. Does this change impact docs?**
- [ X ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by geoffcline** [comment]
https://karpenter-versions.gcline.us/

File: website/content/en/docs/_index.md
Comment (ellistarn) [inline website/content/en/docs/_index.md:3]: Can we think of latest as "undecorated". E.g.

Documentation <------ latest
Documentation (v0.4.0) <--- older
Documentation (v0.3.5) <---- older

File: website/content/en/docs/_index.md
Comment (ellistarn) [inline website/content/en/docs/_index.md:12]: I know this is a joke, but I'd keep this out of PRs so customers don't get confused :)

File: website/content/en/docs/_index.md
Comment (geoffcline) [reply website/content/en/docs/_index.md:3]: I'm worried this would get confusing if we later introduce a "pre-release" section.

docs - pre-release? dev?
docs - latest? current? 
docs - v4
docs - v3

I like the idea of the dropdown showing the current version you are browsing. https://knative.dev/docs/

**Comment by geoffcline** [comment]
take a :eyes: at the dropdown

https://karpenter-versions.gcline.us/docs/

the primary remaining task: create a new hugo page type of "docs-old" that inherits from "docs" but includes a banner saying "this is old" 

see, for ex, https://v1-21.docs.kubernetes.io/docs/home/

File: website/layouts/partials/hooks/body-end.html
Comment (bwagner5) [inline website/layouts/partials/hooks/body-end.html:10]: Should we say something less scary, like "Karpenter ${version} documentation is not the latest stable release. The documentation version you are currently viewing is a static snapshot. ..."

Comment (geoffcline) [reply website/layouts/partials/hooks/body-end.html:10]: Sure

**Comment by geoffcline** [comment]
This is ready for review.

Before merging, the conf.toml must be updated to use *.karpenter.sh urls.

**Comment by geoffcline** [comment]
possible improvements:
- use JS to have "releases" dropdown show the version you are currently browsing.

File: website/config.toml
Comment (ellistarn) [inline website/config.toml:134]: Assuming this will get cleaned up before submitting?

File: website/config.toml
Comment (ellistarn) [inline website/config.toml:141]:
```suggestion
  version = "preview"
```

File: website/config.toml
Comment (ellistarn) [inline website/config.toml:142]:
```suggestion
  url = "https://karpenter-versions.gcline.us/preview-docs/"
```

File: website/config.toml
Comment (geoffcline) [reply website/config.toml:134]: yup! it's just so my preview https://karpenter-versions.gcline.us/docs/ works 

https://github.com/awslabs/karpenter/pull/801#issuecomment-973763810

File: website/layouts/partials/hooks/body-end.html
Comment (ellistarn) [inline website/layouts/partials/hooks/body-end.html:26]: You can share this with the section above if you name the preview version something human readable.
```suggestion
                You are viewing documentation for Karpenter version: ${version}
```

**Review by ellistarn — DISMISSED** [review]
Very cool! Assuming no changes to content otherwise.

File: website/content/en/docs/_index.md
Comment (JacobGabrielson) [reply website/content/en/docs/_index.md]: @geoffcline I like the knative look - what do you think about making it `(Pre-release)` and being slightly grey'd out as they have it?

File: website/layouts/partials/hooks/body-end.html
Comment (geoffcline) [reply website/layouts/partials/hooks/body-end.html:26]: I'd like to keep it separate, because pre-release isn't a "static snapshot"

File: website/config.toml
Comment (geoffcline) [reply website/config.toml:141]: I'd rather keep this as is. Terms are equivalent. difficult to change. Knative and kubernetes use pre-release.

File: website/content/en/docs/_index.md
Comment (geoffcline) [reply website/content/en/docs/_index.md]: I'm good with the structure, we can improve on the style in the future.

File: website/config.toml
Comment (ellistarn) [reply website/config.toml:143]: Knative actually calls it `(Pre-release)` w/ path `development`. https://knative.dev/development/. Not a hill I want to die on.

**Review by ellistarn — APPROVED** [review]

**Review by bwagner5 — APPROVED** [review]
lgtm 🤩

**Review by JacobGabrielson — APPROVED** [review]

**Comment by geoffcline** [comment]
got 2 approvals so I'm merging 🥳

**Pull request merged**
